### PR TITLE
feat(construct): add webcrawler construct

### LINKED ISSUE #291: (Constructs): Web RAG - Web Crawler, Chatting with Web Pages and Search
### Describe the feature

Many RAG experiences are built around websites. Users want to crawl one or more websites, retrieve content from pages, schedule periodic updates, and inject results into OpenSearch to enable RAG requests based on website data.

To support this scenario, a WebCrawler construct can be created. It should be capable of creating new OpenSearch indexes or using existing ones.

This construct can also be used to obtain data from websites in real-time. For example, a user could ask a chatbot to summarize a specific webpage. In this case, the web crawler should extract data from the webpage and provide it to the chatbot.

We should also consider web search scenarios, where users want to use a search engine to obtain results. The results found by the search engine should be parsed and returned to the chatbot.

### Use Case

RAG over websites

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### PR DESCRIPTION
**Pull request by krokoko**
Fixes #291

Add webcrawler construct

---
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the project license.

### CODE REVIEW

File: resources/gen-ai/aws-web-crawler-container/src/common/script-runner.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import { spawn } from 'child_process';
+import { Configuration } from '../managers/config-manager.js';
+
+export abstract class ScriptRunner {
+  public static async downloadFiles(config: Configuration) {
+    return this.execPromise('python3', ['scripts/download_files.py'], {
```
Comment (scottschreckengaust) [inline resources/gen-ai/aws-web-crawler-container/src/common/script-runner.ts:18]: I cannot find this file `download_files.py`

File: resources/gen-ai/aws-web-crawler-container/src/utils.ts [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    const now = new Date();
+    const year = formatNumber(now.getFullYear(), 4);
+    const month = formatNumber(now.getMonth() + 1, 2);
+    const day = formatNumber(now.getDate(), 2);
+    const hours = formatNumber(now.getHours(), 2);
+    const minutes = formatNumber(now.getMinutes(), 2);
+    const seconds = formatNumber(now.getSeconds(), 2);
+    const milliseconds = formatNumber(now.getMilliseconds(), 3);
+
+    const retValue = `${year}-${month}-${day}-${hours}-${minutes}-${seconds}${milliseconds}`;
```
Comment (scottschreckengaust) [inline resources/gen-ai/aws-web-crawler-container/src/utils.ts:63]:
```suggestion
    const retValue = `${now.toISOString()}`;
```

Comment (scottschreckengaust) [inline resources/gen-ai/aws-web-crawler-container/src/utils.ts:63]:
```suggestion
    const now = new Date();
    const retValue = `${now.toISOString()}`;
```

File: lambda/aws-web-crawler-scheduler/lambda.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
+import os
```
Comment (dineshSajwan) [inline lambda/aws-web-crawler-scheduler/lambda.py:1]: Please add Copyright and license comments at the top of file.

File: lambda/aws-web-crawler-lambda/index.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+ *
+ *  or in the 'license' file accompanying this file. This file is distributed on an 'AS IS' BASIS, WITHOUT WARRANTIES
+ *  OR CONDITIONS OF ANY KIND, express or implied. See the License for the specific language governing permissions
+ *  and limitations under the License.
+ */
+import 'jquery';
+import { Crawler } from './crawler';
+import { CrawlerRequestTest } from './types';
+
+/*
```
Comment (dineshSajwan) [inline lambda/aws-web-crawler-lambda/index.ts:17]: please remove this commented urls.

File: resources/gen-ai/aws-web-crawler-container/scripts/download_files.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
+import os
```
Comment (dineshSajwan) [inline resources/gen-ai/aws-web-crawler-container/scripts/download_files.py:1]: copyright & license comment.

**Review by dineshSajwan — APPROVED** [review]
Few nitpicks.. other wise LGTM ..

File: resources/gen-ai/aws-web-crawler-container/scripts/download_files.py (hunk 5 above)
Comment (krokoko) [reply resources/gen-ai/aws-web-crawler-container/scripts/download_files.py:13]: fixed in https://github.com/awslabs/generative-ai-cdk-constructs/pull/474/commits/94869bc149e4ed35dee85ff8a903754189564663

File: lambda/aws-web-crawler-lambda/index.ts (hunk 4 above)
Comment (krokoko) [reply lambda/aws-web-crawler-lambda/index.ts]: Fixed in https://github.com/awslabs/generative-ai-cdk-constructs/pull/474/commits/94869bc149e4ed35dee85ff8a903754189564663

File: lambda/aws-web-crawler-scheduler/lambda.py (hunk 3 above)
Comment (krokoko) [reply lambda/aws-web-crawler-scheduler/lambda.py:13]: Fixed in https://github.com/awslabs/generative-ai-cdk-constructs/pull/474/commits/94869bc149e4ed35dee85ff8a903754189564663

File: resources/gen-ai/aws-web-crawler-container/src/utils.ts (hunk 2 above)
Comment (krokoko) [reply resources/gen-ai/aws-web-crawler-container/src/utils.ts:63]: ToISOString > "2024-06-10T18:44:21.978Z"
Existing code > "2024-06-10-13-44-21978"

Comment (krokoko) [reply resources/gen-ai/aws-web-crawler-container/src/utils.ts]: Fixed in [475de44](https://github.com/awslabs/generative-ai-cdk-constructs/pull/474/commits/475de44a683aed19f7e07033e48bd82972e099eb)

**Pull request merged**